docs/learn/multi-format-reader.md

```diff
@@ -0,0 +1,263 @@
+
+In addition, the reader can also implement the `get_data_dims` method, which is used to return a list of the data dimensions (see below for more details).
+
+All of `get_attr`, `get_data`, and `get_eln_data`  (as well as any similar method that might be implemented) should have the same call signature:
+```python
+def get_data(self, key: str, path: str) -> Any:
+```
+Here, `key` is the config dict key (e.g., `"/ENTRY[my-entry]/data/data"`) and path is the path that comes _after_ the prefix in the config file. In this example, `path` would be `mydata`. With these two inputs, the reader should be able to return the correct data for this template key.
+
+### Special rules
+- **Lists as config value**: It is possible to write a list of possible configurations of the sort
+  ```json
+  "/ENTRY/title":"['@attrs:my_title', '@eln', 'no title']"
+  ```
+  The value most be a string of a list, with each item being a string itself. This allows to provide different options depending if the data exists for a given callback. For each list item , it is checked if a value can be returned and if so, the value is written. In this example, the converter would check (in order) the `@attrs` (with path `"my_title"`) and `@eln` (with path `""`) tokens and write the respective value if it exists. If not, it defaults to "no title".
+  This concept can be particularly useful if the same config file is used for multiple measurement configurations, where for some setup, the same metadata may or not be available.
+
+    Note that if this notation is used, it may be helpful to pass the `suppress_warning` keyword as `True` to the read function. Otherwise, there will be a warning for every non-existent value.
+
+- **Wildcard notation**: There exists a wildcard notation (using `*`)
+  ```json
+  "/ENTRY/data/AXISNAME[*]": "@data:*.data",
+  ```
+  that allows filling multiple fields of the same type from a list of dimensions. This can be particularly helpful for writing `DATA` and `AXISNAME` fields that are all stored under similar paths in the read-in data.
+  For this, the `get_data_dims` method need to implemented. For a given path, it should return a list of all data axes available to replace the wildcard.
+    
+    The same wildcard notation can also be used with a name to repeat entries with different names (e.g., *{my, name, etc} replaces to three keys with my, name, etc replaced, respectively). As an example, for multiple lenses and their voltage readouts, one could write:
+  ```json
+  "LENS_EM[lens_*{A,B,C,Foc}]": {
+    "name": "*",
+    "voltage": "@attrs:metadata/file/Lens:*:V",
+    "voltage/@units": "V"
+  },
+  ```
+  which would write `NXlens_em` instances named `lens_A`, `lens_B`, and `lens_Foc`.
+
+    Note that the `get_data_dims` must correspond to all use cases, i.e., for `@data`, `@attrs`, and so on.

```

Comment:
This I don't understand. Will this not just replace "*" by the entries in the dict? Does this use case really require get_data_dims?

Reply:
removed